clash-prelude/src/Clash/Signal/Internal.hs

```diff
@@ -1276,7 +1276,8 @@ unsafeFromReset (Reset r) = r
 -- __NB__: You probably want to use 'unsafeFromActiveLow' or
 -- 'unsafeFromActiveHigh'.
 unsafeToReset
-  :: Signal dom Bool
+  :: KnownDomain dom
```

Comment:
The `KnownDomain` constraint is unused. Isn't GHC technically allowed to prune it?

Reply:
I think that's okay here. As I understand it, this is purely to make the API change now already for 1.8.0 so we can change the implementation later on. So only the user-facing API is what matters here, GHC can prune it without problems. Once the underlying code changes, the constraint will be needed, and it will no longer be pruned.

Huh. Even in general I think it's okay if it is pruned. The constraint is only needed in /Haskell/, and GHC can do what it wants as long as the code continues to work. We worry about stuff being pruned because we'd like to access it during HDL generation, but HDL generation doesn't need the `KnownDomain` for this function, it's a wire, nothing more.

Reply:
Ah, yes, you're right